src/main/java/winstone/Launcher.java

```diff
@@ -208,6 +214,22 @@ public Launcher(Map args) throws IOException {
         Runtime.getRuntime().addShutdownHook(new ShutdownHook(this));
     }
 
+    private void writePortToFileIfNeeded() throws IOException {
+        String portFileName = System.getProperty(WINSTONE_PORT_FILE_NAME_PROPERTY);
+        if (portFileName != null) {
+            Connector connector = server.getConnectors()[0];
+            if (connector instanceof ServerConnector) {
+                int port = ((ServerConnector) connector).getLocalPort();
+                File portFile = new File(portFileName);
+                File portDir = portFile.getParentFile();
+                portDir.mkdirs();
+                try (BufferedWriter writer = new BufferedWriter(new FileWriter(portFile))) {
+                    writer.write(Integer.toString(port));
```

Comment:
This is using the platform default charset - but [downstream ](https://github.com/jenkinsci/acceptance-test-harness/pull/711/files#diff-d209b411ccb684e771a7622ed22cfc92b82abf6e5954e272bee7d6e8edb1ffe3R80) PRs are assuming `UTF-8`

Reply:
it's a number so not sure charset matters :)